ads/alp/handler.js

```diff
@@ -181,3 +199,51 @@ export function warmupDynamic(e) {
 function getHeadOrFallback(doc) {
   return doc.head || doc.documentElement;
 }
+
+/**
+ * Returns info about an ancestor that can perform A2A navigations
```

Comment:
I think the doc is correct. Currently it is always the 2nd top ancestor, but one of the TODO is to also allow the top document. The latter is needed for webview based viewer.